src/world/entity.rs

```diff
@@ -331,6 +346,39 @@ impl<'a> Join for &'a EntitiesRes {
 
 unsafe impl<'a> ParJoin for &'a EntitiesRes {}
 
+/// An entity builder from `EntitiesRes`.  Allows building an entity with its
+/// components if you have mutable access to the component storages.
+pub struct EntityResBuilder<'a> {
+    /// The entity being built
+    pub entity: Entity,
```

Comment:
Because this is public you can change it at will. So you could partially build entity, switch it another one, forget build it and then only the switched one gets removed.

There is nothing horribly wrong with this, but it seems to just lead to confusing situations. I would make entity private and create getter for it.

Reply:
It also is bit confusing that building doesn't actually do anything. Normally with builders this cannot be observed, but because you can switch entity you can with this.